mcp23xxx/group.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2025 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package mcp23xxx
+
+import (
+	"fmt"
+	"time"
+
+	"periph.io/x/conn/v3"
+	"periph.io/x/conn/v3/gpio"
+	"periph.io/x/conn/v3/pin"
+)
+
+// The internal structure for a group of pins.
+type pinGroup struct {
+	dev         *Dev
+	port        int
+	pins        []*portpin
+	defaultMask gpio.GPIOValue
+}
+
+// Group returns a gpio.Group that is made up of the specified pins.
+func (dev *Dev) Group(port int, pins []int) *gpio.Group {
+	grouppins := make([]*portpin, len(pins))
+	for ix, number := range pins {
+		pp, ok := dev.Pins[port][number].(*portpin)
+		if !ok {
+			return nil
+		}
+		grouppins[ix] = pp
+	}
+	defMask := gpio.GPIOValue((1 << len(pins)) - 1)
+	var pgif interface{} = &pinGroup{dev: dev, port: port, pins: grouppins, defaultMask: defMask}
+	if gpiogroup, ok := pgif.(gpio.Group); ok {
+		return &gpiogroup
+	}
+
+	return nil
+}
+
+// Pins returns the set of pin.Pin that make up that group.
+func (pg *pinGroup) Pins() []pin.Pin {
```

Comment:
One thing I noticed: since gpio.Group is about gpio.Pin, should the methods of the interfaces return gpio.PinIO instead pin.Pin? This would remove a lot of casting for the end user.

Reply:
PinIO is both read and write. There are some devices that don't support read.

Reply:
Yeah I understand the trade off. It's okay as is.